warp10/src/main/java/io/warp10/continuum/gts/MetadataSelectorMatcher.java

```diff
@@ -0,0 +1,178 @@
+package io.warp10.continuum.gts;
+
+
+import io.warp10.continuum.store.thrift.data.Metadata;
+import io.warp10.script.WarpScriptException;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.text.ParseException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MetadataSelectorMatcher {
+
+  private final String selector;
+
+  private final Matcher classnamePattern;
+  private final boolean withClassnameSelector;
```

Comment:
the withXXX variables are not needed, simply check for nullity of the assocaited pattern variables